packages/@aws-cdk/aws-codeartifact/lib/repository.ts

```diff
@@ -0,0 +1,404 @@
+  public grantReadWrite(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [...actions.readActions, ...actions.writeActions],
+      resourceArns: [this.repositoryArn],
+    });
+  }
+
+  /**
+     * Defines a CloudWatch event rule which triggers for repository events. Use
+     * `rule.addEventPattern(pattern)` to specify a filter.
+     */
+  public onEvent(id: string, options: events.OnEventOptions = {}) {
+    const rule = new events.Rule(this, id, options);
+    rule.addEventPattern({
+      source: ['aws.codeartifact'],
+      detail: {
+        domainName: [this.repositoryDomainName],
+        domainOwner: [this.repositoryDomainOwner],
+        repositoryName: [this.repositoryName],
+      },
+    });
+    rule.addTarget(options.target);
+    return rule;
+  }
+
+  /**
+     * Defines a CloudWatch event rule which triggers when a "CodeArtifact Package
+     *  Version State Change" event occurs.
+     */
+  public onPackageVersionStateChange(id: string, options: events.OnEventOptions = {}): events.Rule {
+    const rule = this.onEvent(id, options);
+    rule.addEventPattern({
+      detailType: ['CodeArtifact Package Version State Change'],
+    });
+    return rule;
+  }
+}
+
+/**
+ * A new CodeArtifacft repository
+ */
+export class Repository extends RepositoryBase {
+  /**
+     * Import an existing Repository provided an ARN
+     *
+     * @param scope The parent creating construct
+     * @param id The construct's name
+     * @param repositoryArn repository ARN (i.e. arn:aws:codeartifact:us-east-2:444455556666:repository/my-domain/my-repo)
+     */
+  public static fromRepositoryArn(scope: Construct, id: string, repositoryArn: string): IRepository {
+    const parsed = Stack.of(scope).parseArn(repositoryArn);
+    const spl = parsed.resourceName?.split('/') || ['', ''];
+    const repositoryName = spl[1];
+    const domainName = spl[0];
+
+    class Import extends RepositoryBase {
+      public repositoryDescription: string = '';
+      public repositoryName: string = repositoryName || '';
+      public repositoryDomainOwner: string = parsed.account || '';
+      public repositoryDomainName: string = domainName || '';
+      public readonly repositoryArn = repositoryArn;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly repositoryArn: string;
+  public readonly repositoryName: string;
+  public readonly repositoryDomainOwner: string;
+  public readonly repositoryDomainName: string;
+  public readonly repositoryDescription: string;
+  private readonly cfnRepository: ca.CfnRepository;
+
+  constructor(scope: Construct, id: string, props: RepositoryProps) {
+    super(scope, id);
+
+    this.cfnRepository = new ca.CfnRepository(this, id, {
+      repositoryName: props.repositoryName,
+      description: props.description,
+    });
+
+    this.cfnRepository.addPropertyOverride('DomainName', props.domain.cfnDomain.attrName);
```

Comment:
Seriously? How is this missing? Please add a comment at the least, explaining that this should be included in the props soon.

Reply:
This was not in the CFN L1 version so this line was added. It's required so it should be present.

Reply:
Yeah, I gotcha that the CFN L1 is missing it. That's what shocks me. I think adding a comment that along the effect of ` // This should be added to the L1 props soon, but until then this is required to create a Repository` or something like that just to drive home that we should be on the look-out for the "fix" from CFN soon.

Reply:
I've added this comment. I was shocked as well, and a bit perplexed/frustrated when this came up. We should escalate this to find out what happened.

Reply:
See above comment from `domain.ts` about missing properties that will be updated in the next documentation release. This aspect has been resolved.